.github/workflows/ci.yml

```diff
@@ -13,9 +13,59 @@ env:
   API_CLIENT_SECRET: clientsecret
 
 jobs:
-  test:
+  type_checking:
+    name: "Type check 🔎"
     runs-on: ubuntu-latest
+    steps: 
+      - name: Check out code
+        uses: actions/checkout@v4.1.7
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v4.0.3
+        with:
+          node-version-file: '.node-version'
+          cache: 'npm'
+
+      - name: Installing dependencies
+        run: npm install       
+
+      - name: Building source
+        run: npm run build
+     
+      - name: Running Lint checks
+        run: npm run lint
+    
+      - name: Pulling the latest type from the API repo
+        run: npm run generate-types
```

Comment:
I still think this shouldn't be run on every test, as we have separate PRs for updating those types, so now feels like a good time to remove it

Reply:
Lets keep this in for the moment, I think now that its separated into its own job we can easily see that the unit/integrations tests pass and until the types are updated a specific PR will be blocked. Its als osuper quick. We dont have to then mark a PR ready for review, get it approved and then have to put it back into draft so that it doesn't accidentally get merged

Reply:
Lets keep this in for the moment, I think now that its separated into its own job we can easily see that the unit/integrations tests pass and until the types are updated a specific PR will be blocked. Its als osuper quick. We dont have to then mark a PR ready for review, get it approved and then have to put it back into draft so that it doesn't accidentally get merged